### PR TITLE
Test convex verify and mutate session

### PR DESCRIPTION
**Pull request by ncovercash**
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "dotenv": "^16.3.1",
     "drizzle-orm": "^0.29.1",
     "jotai": "^2.6.0",
+    "ky": "^1.2.0",
```
Comment (ncovercash) [inline package.json:45]: fetch but ✨ better ✨

File: convex/verifyAndMutateSession.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import type { AppRouter } from '@/server/routers/_app';
```
Comment (ncovercash) [inline convex/verifyAndMutateSession.ts:1]: smaller files = easier life = easier testing

File: convex/verifyAndMutateSession.test.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      json: () => Promise.resolve(data),
+    });
+  }
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it.each([
+    undefined,
```
Comment (ncovercash) [inline convex/verifyAndMutateSession.test.ts:19]: makes it super easy to run similar test cases

**Comment by kalkafox** [comment]
Thanks again!

**Pull request merged**